### DISCUSSION TITLE
Convert nodes into HTML format

### DISCUSSION THREAD

**Discussion by charklewis** [facebook/lexical#1941] [Q&A] [answered]
This might be a naive question, is there a way to convert an editors state into HTML elements programmatically? I am thinking something like what slateJS has - https://docs.slatejs.org/concepts/10-serializing#html. I can see this being a potential issue with custom decorator nodes. 

I have tried `editorState.read(() => $getRoot().exportDOM(editor))` but either see nothing returned, or `createDOM: base method not extended` depending on where I call it. 

My user case for this is rendering the content in the database (which has been saved as `JSON.stringify(editorState.toJSON())`) and rendering it as a read only component (somewhere else in the app that has nothing to do with lexical or editing the content). 

I have achieved this at the moment by rendering a "Preview" version of my `LexicalComposer`, and wondering if I can do this by just rendering HTML.

**Comment by zurfyx** [comment, marked as answer]
Not really an answer but it's a work in progress -> https://github.com/facebook/lexical/issues/1834

The fundamentals exists (as private functions) inside @lexical/clipboard but they have to structured in a proper way so that we can offer an ergonomic API that beyond copy-paste
